### DISCUSSION TITLE
Can I set CONFIG_TICKLESS_KERNEL config to each core?

### DISCUSSION THREAD

**Discussion by chojpsh1** [zephyrproject-rtos/zephyr#78195] [Q&A] [answered]
Hello, I have a question about CONFIG_TICKLESS_KERNEL.
I wonder if CONFIG_TICKLESS_KERNEL can be set differently for each core (Activate the config only for specific cores)
I'd appreciate if you could reply. Thanks!

**Comment by ndrs-pst** [comment]
It looks like you want TICKLESS on one core and non-TICKLESS on another.
-> Before discussing whether this is possible or not, I’m curious about the specific challenge you're looking to address. 😃

**Reply by chojpsh1** [reply]
I apologize for the late response. You are correct in what you have mentioned. 
The product we are developing is a 4-core device with an ARM architecture timer that operates per core. As an example, we want to fix our self-developed watchdog function on Core 1 and make the remaining cores operate as TICKLESS_KERNEL except for Core 1.
I was curious if this would be possible.

**Comment by ndrs-pst** [comment, marked as answer]
Thank you for your clarification.
As far as I know, Zephyr has a global scheduler regardless of how many cores the CPU has.
This implies that you can only select between TICKLESS or non-TICKLESS, but not both.

Hence, the suggestion might look like this:
- `CONFIG_TICKLESS_KERNEL=y`
- Use [Timer APIs](https://docs.zephyrproject.org/apidoc/3.7.0/group__timer__apis.html) to periodically trigger a specified thread that has been pinned to CORE1 via [`k_thread_cpu_pin`](https://docs.zephyrproject.org/apidoc/3.7.0/group__thread__apis.html#gae9ebd9845e14ed02944ab9282a185c03).

---
A resource worth looking at.
- https://docs.zephyrproject.org/3.7.0/kernel/services/scheduling/index.html
- https://docs.zephyrproject.org/3.7.0/kernel/services/smp/smp.html

**Reply by chojpsh1** [reply]
Thank you for your detailed explanation, it was really helpful. Have a great day!
